### PR TITLE
Set/unset presence

### PR DESCRIPTION
**Pull request by MaximilianKrambach**
A simple view (meant to be operated in combination with a bar code scanner filling in the auto focus field and sending 'return') to set unset the 'is_present' status for a user based on user number, displaying the result of the last action.

Overall permissions to access this url are broken (user-routing), so the view is empty if permissions are wrong

(my IDE did some heavy indentation fix, which I commited separately)

### CODE REVIEW

**Comment by emanuelschuetze** [comment]
Looks good. But the auto focus for the input field does not work. Can you check it please?

File: client/src/app/site/users/services/user-repository.service.ts [hunk 1]
```diff
@@ -232,6 +234,26 @@ export class UserRepositoryService extends BaseRepository<ViewUser, User> {
         return results;
     }
 
+    /**
+     * Searches and returns Users by full name
+     *
+     * @param number: A participant number
+     * @returns all users matching that number
+     */
+    public getUsersByNumber(number: string): ViewUser[] {
```
Comment (FinnStutzenstein) [inline client/src/app/site/users/services/user-repository.service.ts]: Whats about `return this.getViewModelList().filter(user => user.participant_number === number);`?

**Comment by MaximilianKrambach** [comment]
autofocus should be fixed now @emanuelschuetze

File: client/src/app/site/users/presence-detail/presence-detail.component.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    /**
+     * Constructor. Empty
+     */
+    public constructor(
+        private userRepo: UserRepositoryService,
+        private formBuilder: FormBuilder,
+        private operator: OperatorService,
+        config: ConfigService
+    ) {
+        config.get('users_enable_presence_view').subscribe(conf => (this._enabledInConfig = conf));
```
Comment (tsiegleauq) [inline client/src/app/site/users/presence-detail/presence-detail.component.ts]: documentation missing. Constructor is not empty

File: client/src/app/site/users/services/user-repository.service.ts [hunk 3]
```diff
@@ -232,6 +234,16 @@ export class UserRepositoryService extends BaseRepository<ViewUser, User> {
         return results;
     }
 
+    /**
+     * Searches and returns Users by full name
```
Comment (tsiegleauq) [inline client/src/app/site/users/services/user-repository.service.ts]: Searches and returns users by their participant number

File: client/src/app/site/users/presence-detail/presence-detail.component.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    /**
+     * Triggers the user finding and updating process. The user number will be taken from the {@link userForm}.
+     * Feedback will be relayed to the {@link errorMsg} and/or {@link lastChangedUser} variables
+     */
+    public async changePresence(): Promise<void> {
+        const number = this.userForm.get('number').value;
+        const users = this.userRepo.getUsersByNumber(number);
+        this.userForm.setValue({ number: null });
```
Comment (tsiegleauq) [inline client/src/app/site/users/presence-detail/presence-detail.component.ts]: `this.userForm.reset()`
Also removes "touched" events and possible validators

**Comment by MaximilianKrambach** [comment]
done @tsiegleauq

**Comment by MaximilianKrambach** [comment]
done @tsiegleauq

**Pull request merged**